datamade/la-metro-councilmatic
# Update metro districts

## Overview

Adds in the new LA Metro boundaries that changed after redistricting.

### Checklist


### Notes

#### Issues

Initially, I tried to just swap in the URL for the new boundaries (remotely stored in a secret Gist by @hancush) in the Makefile. However, this caused two issues:

1. The properties in the new boundaries do not match the properties in the GeoJSON files that the Makefile currently fetches from the DataMade OCD endpoint. In particular, the new boundaries use `LABEL` while the old files use `name`. This caused `add_district.py` to fail while labeling the features.
2. The new boundaries are structured as a nested `FeatureCollection`. The actual boundaries are inside `FeatureCollection` that is essentially empty. This caused `combine.py` to add an empty `FeatureCollection` to the final boundaries instead of the new Metro boundaries.

#### Current Solutions

I manually modified the new boundaries (stored as `supervisory_districts.geojson`) to work with the current scripts. Because the data pipeline now depends on a local file, inline with DataMade's ETL guidelines,  I added the directory `data/raw/` to store both `supervisory_districts.geojson` and the files fetched from the OCD endpoint. I also added `data/final/` to store the final output boundaries, `boundary.geojson`, that are displayed on the site. 

The new directories and use of `supervisory_districts.geojson` necessitated minor changes in the Makefile. Additionally, I added a `clean` command to clear the `data/final/` if we want to re-run the Makefile and generate `data/final/boundary.geojson`.

Both `supervisory_districts.geojson` and the data fetched from the OCD endpoint are stored in `data/raw/` but if it'd be more appropriate to separate those (and perhaps add a command to clean the fetched files), please let me know.

#### Docker

Currently, Docker isn't able to run `make`. For some reason, the container run by `docker-compose` doesn't have `make` installed. My suspicion is that this could be due some aspect of the Dockerfile. I attempted to modify it to get access to make but didn't have any luck. 

We can discuss whether or not we want Docker to be able to run `make` and how to get that done.

## Testing Instructions

 * Run `make clean` and then `make all` from the `data/` directory.
 * Import the new shapes with `docker-compose run --rm app python manage.py import_shapes data/final/boundary.geojson`
 * Verify that the new boundaries are live (might need to clear your cache)

Handles #799
